Polygon.js

```diff
@@ -0,0 +1,129 @@
+{
+	"translatorID": "fa7c37b1-fda4-418a-a8b8-2491929411ab",
+	"label": "Polygon",
+	"creator": "czar",
+	"target": "^https?://(?:www\\.)polygon\\.com",
+	"minVersion": "1.0",
+	"maxVersion": "",
+	"priority": 100,
+	"inRepository": true,
+	"translatorType": 4,
+	"browserSupport": "g",
+	"lastUpdated": "2015-12-06 02:04:51"
+}
+
+// http://en.wikipedia.org/wiki/User_talk:Czar
+
+function detectWeb(doc, url) {
+	var pageType = ZU.xpathText(doc, '//script[contains(., "page_type :")]').split('page_type : \"')[1].split(/\"/)[0];
+	if (pageType == "Article") { return "webpage" } 
+	return null;
+}
+
+function scrape(doc, url) {
+	var newItem = new Zotero.Item("webpage");
+	newItem.websiteTitle = "Polygon";
+	newItem.language = "en-US";
+
+	newItem.url = doc.location.href;
+	newItem.title = ZU.xpathText(doc, '//h2[@class="m-entry__title"]');
+	newItem.date = ZU.xpathText(doc, '//meta[@property="article:published_time"]/@content').split("T")[0];
+
+	newItem.abstractNote = ZU.xpathText(doc, '//meta[@name="description"]/@content');
+	newItem.notes = [ ZU.xpathText(doc, '//*[@id="article-body"]').trim() ];
+	newItem.attachments.push({document:doc, title:"Polygon snapshot" ,mimeType:"text/html"});
+
+		// Authors
+	var authors = ZU.xpathText(doc, '//script[contains(., "dataLayer =")]').split('\"Author\":\"')[1].split(/\",/)[0].split(":");
```

Comment:
@zuphilip - any particular reason? I personally prefer using `match` over
`split`, too, but I don't see any reason to reject split for this. But I
may be overlooking something?

On Sun, Dec 6, 2015 at 10:18 AM, Philipp Zumstein <notifications@github.com>
wrote:

> In Polygon.js
> <https://github.com/zotero/translators/pull/973#discussion_r46772348>:
>
> > +
> > +function scrape(doc, url) {
> > +	var newItem = new Zotero.Item("webpage");
> > +	newItem.websiteTitle = "Polygon";
> > +	newItem.language = "en-US";
> > +
> > +	newItem.url = doc.location.href;
> > +	newItem.title = ZU.xpathText(doc, '//h2[@class="m-entry__title"]');
> > +	newItem.date = ZU.xpathText(doc, '//meta[@property="article:published_time"]/@content').split("T")[0];
> > +
> > +	newItem.abstractNote = ZU.xpathText(doc, '//meta[@name="description"]/@content');
> > +	newItem.notes = [ ZU.xpathText(doc, '//*[@id="article-body"]').trim() ];
> > +	newItem.attachments.push({document:doc, title:"Polygon snapshot" ,mimeType:"text/html"});
> > +
> > +		// Authors
> > +	var authors = ZU.xpathText(doc, '//script[contains(., "dataLayer =")]').split('\"Author\":\"')[1].split(/\",/)[0].split(":");
>
> I suggest also to use a regexp to extract the relevant part, something
> like match('/"Author"\s?:\s?"([^"]*)"/')
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/zotero/translators/pull/973/files#r46772348>.
>



-- 
Sebastian Karcher, PhD
www.sebastiankarcher.com